### PR TITLE
GG-29788 Unclear regression in 2.5.22 and 8.7.20

### PR DESCRIPTION
**Pull request by korlov42**

### CODE REVIEW

File: modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/ConcurrentStripedPool.java [hunk 1]
```diff
@@ -123,37 +123,29 @@ public void clear() {
...
-                    return it.hasNext();
                 }
-                else
-                    return false;
+
+                return false;
             }
 
             @Override public E next() {
                 if (it.hasNext())
```
Comment (AMashenkov) [inline modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/ConcurrentStripedPool.java:145]: Should "it.hasNext()" be changed to "hasNext()" here?

File: modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/ConnectionManager.java [hunk 2]
```diff
@@ -224,10 +224,8 @@ public void stop() {
      * Called periodically to clean up the statement cache.
      */
     private void cleanupStatements() {
-        long now = U.currentTimeMillis();
-
         connPool.forEach(c -> {
-            if (now - c.statementCache().lastUsage() > stmtTimeout)
+            if (c.statementCache().inactiveFor(stmtTimeout))
```
Comment (AMashenkov) [inline modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/ConnectionManager.java:228]: Let's convert stmtTimeout to nanos.

File: modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/ConcurrentStripedPool.java (hunk 1 above)
Comment (korlov42) [reply modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/ConcurrentStripedPool.java:145]: it MUST, thx

File: modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/ConnectionManager.java (hunk 2 above)
Comment (korlov42) [reply modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/ConnectionManager.java:228]: fixed

**Pull request merged**